service/topology_coordinator.cc

```diff
@@ -312,6 +315,10 @@ class topology_coordinator : public endpoint_lifecycle_subscriber {
         }
 
         if (!next_req) {
+            if (!topo.paused_requests.empty()) {
+                // Allow paused requests to finish.
+                return std::move(guard);
+            }
```

Comment:
Why is it fine to not cancel requests that are not paused in this case? Why can't we cancel not paused requests while waiting for paused requests?